### PR TITLE
Fix deriveSponsorWallet function

### PR DESCRIPTION
**Pull request by Siegrift**
Closes https://github.com/api3dao/commons/issues/60

### CODE REVIEW

File: src/blockchain-utilities/derivation.test.ts [hunk 1]
```diff
@@ -83,11 +84,27 @@ describe('deriveWalletPathFromSponsorAddress', () => {
     it('throws deriving a sponsor wallet due to an invalid DApi name', () => {
       const dapiName = 'invalid dapi name';
       const throwingFn = () =>
-        deriveSponsorWallet('test test test test test test test test test test test junk', dapiName);
+        deriveSponsorWallet('test test test test test test test test test test test junk', dapiName, 'AIRSEEKER');
```
Comment (Siegrift) [inline src/blockchain-utilities/derivation.test.ts:87]: I am not a huge fan of these uppercased protocol names, but don't want users to pass numbers. If this is lowercased, so should be `PROTOCOL_IDS` below.

Comment (dcroote) [reply src/blockchain-utilities/derivation.test.ts:87]: To me, uppercase suggests they represent more than any old string, which is a benefit. Uppercase also matches Airnode:

https://github.com/api3dao/airnode/blob/707491e6a54d94dfbda56f26d4be2e7504a945e4/packages/airnode-protocol/src/index.ts#L39-L46

File: src/blockchain-utilities/derivation.ts [hunk 2]
```diff
@@ -75,17 +79,15 @@ export const fromBytes32String = (input: string) => ethers.utils.parseBytes32Str
 /**
  * Derives a sponsor wallet, given a mnemonic and dapiName.
  *
- * @param sponsorWalletMnemonic the sponsor wallet mnemonic
+ * @param airnodeMnemonic the sponsor wallet mnemonic
```
Comment (dcroote) [inline src/blockchain-utilities/derivation.ts:82]: Update "the sponsor wallet mnemonic"? Perhaps for completeness, add the new `protocolName` `@param`?

File: src/blockchain-utilities/derivation.test.ts (hunk 1 above)
Comment (Siegrift) [reply src/blockchain-utilities/derivation.test.ts:87]: Ok. Leaving it as is then.

File: src/blockchain-utilities/derivation.ts (hunk 2 above)
Comment (Siegrift) [reply src/blockchain-utilities/derivation.ts:82]: Thanks. Although I am not sure if this documentation is not an overkill. Seems like all of what's written can be derived from the parameter names/types.

**Review by andreogle — APPROVED** [review]

File: src/blockchain-utilities/derivation.test.ts (hunk 1 above)
Comment (Siegrift) [reply src/blockchain-utilities/derivation.test.ts:87]: I realized that we would be mixing `protocol ID` and `protocol name` in some functions, so I sticked with using the protocol ID everywhere.

File: src/blockchain-utilities/derivation.test.ts [hunk 3]
```diff
@@ -75,20 +75,37 @@ describe('deriveWalletPathFromSponsorAddress', () => {
       const dapiName = ethers.utils.formatBytes32String('BTC/ETH');
       const sponsorWallet = deriveSponsorWallet(
         'test test test test test test test test test test test junk',
-        dapiName
+        dapiName,
+        '1'
       );
 
-      expect(sponsorWallet.address).toBe('0x1e6f0dfb1775f5032f12f56a01526351eD3F07aF');
+      expect(sponsorWallet.address).toBe('0x4f86228e0Bc58829Cd77547224291bb8d212174D');
```
Comment (Siegrift) [inline src/blockchain-utilities/derivation.test.ts:82]: There was a bug in this function before.

**Pull request merged**